### ISSUE TITLE
[FEATURE] Persist -detailed-exitcode terraform flag with terrmate

### ISSUE THREAD

**Issue by cvr12** [terramate-io/terramate#2016]
**Is your feature request related to a problem? Please describe.**
When using -detailed-exitcode for terraform plan (see [terraform docs](https://developer.hashicorp.com/terraform/cli/commands/plan#detailed-exitcode) terramate when there are changes in the plan (terraform returns exit code 2) since this is non-zero 

```
Error: one or more commands failed
> execution failed: running /opt/homebrew/bin/terraform plan -detailed-exitcode -out keikaku (in {DIRECTORY}): exit status 2
```

**Describe the solution you'd like**
Any solution that means that the outcome of running the terramate command means the terraform plan exit code is persisted so it can be used in a pipeline / script.

**Describe alternatives you've considered**
Tried to use --detailed-exit-code flag (seems this is only supported on generate)

**Additional context**

**Comment by mar-pan** [comment]
bump

**Comment by mariux** [comment]
Thanks for suggesting this. What is Terramate expected to do here? 

You can wrap Terraform into a script and act on the exit code right after it finished execution or write away the results to postpone any action. 

You would then run your script instead of the terraform binary.

**Comment by matphilippe** [comment]
> Thanks for suggesting this. What is Terramate expected to do here?
> 
> You can wrap Terraform into a script and act on the exit code right after it finished execution or write away the results to postpone any action.
> 
> You would then run your script instead of the terraform binary.

That sounds very reasonable, thanks for the suggestion, it actually unblocked my use-case.
I am sharing how, in case it helps someone else.

I am willing to contribute a cleaner version to the docs if that's relevant.

My goal was to know which are the stacks whose terraform plan show an actual diff, and wether or not there is a failing plan.
This is for CI purposes.

The solution I found starts with this:

```bash
terramate run --changed -- terraform init -lock-timeout=5m
terramate run --changed -- task plan-one
```

where _task plan-one_ does

```bash
terraform plan -out out.tfplan -lock-timeout=5m -detailed-exitcode || echo $? > "plan_exit_code"
``` 

(We are using [TaskFile](https://taskfile.dev/), but could do without)

From there, we can easily inspect the stacks for exit codes, and run whathever logic seems adequate.
